dune/iga/geometrykernel/slicecurve.hh

```diff
@@ -0,0 +1,82 @@
+// SPDX-FileCopyrightText: 2023 The dune-iga developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#pragma once
+
+#include <dune/iga/splines/bsplinealgorithms.hh>
+#include <dune/iga/splines/nurbsalgorithms.hh>
+
+namespace Dune::IGANEW {
+
+  // Inspired by https://github.com/pradeep-pyro/tinynurbs/blob/master/include/tinynurbs/core/modify.h
+  template <typename GeoCurve, typename ScalarType = typename GeoCurve::ctype> requires (GeoCurve::mydimension == 1)
+  auto splitCurve(const GeoCurve& geoCurve, ScalarType u) -> std::pair<GeoCurve, GeoCurve> {
+    using ControlPointType = typename GeoCurve::ControlPointType;
+    using PatchData = typename GeoCurve::PatchData;
+    using ControlPointNetType = typename GeoCurve::ControlPointNetType;
+
+    const PatchData& patchData     = geoCurve.patchData();
+    auto knots          = patchData.knotSpans.front();
+    auto domain         = geoCurve.domain();
+    auto degree         = patchData.degree[0];
+
+    const int span = Splines::findSpan(degree, u, knots);
+    const int multiplicity = std::ranges::count_if(knots, [=](double val) { return u == val ; });
```

Comment:
In line 23 you already compute with `findSpan`, the count_if goes through the whole vector again. I added a function to help with this. I hope I don break everaything again